diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -390,9 +390,11 @@ namespace diff_drive_controller
     }
 
     // Publish odometry message
-    if (last_state_publish_time_ + publish_period_ < time)
+    const ros::Duration half_period(0.5 * period.toSec());
```

Comment:
I have to say two things about this:

1. I couldn't do `if (last_state_publish_time_ + publish_period_ < time + 0.5 * period)` because `period` is a const reference and `ros::Duration` only have the `operator*` defined as a method modifying `this`. :dizzy_face:  

2. The `period`, although very similar between `update` calls, it's not a constant and provides the time since the last call to `update`. I also don't know if there's something that provides the update frequency in a clean way. I didn't see anything on ROS control API.

That said, I don't see where do you suggest to move this instruction? Do you mean on the `init` method? I don't have the `period` there and I don't know how to find it. Furthermore, the `period` isn't constant.
If (for some reason) the `period` is very large after one call, this implementation would assume the next call would take the same `period`, which I think in this case is a good and safe assumption.